### ISSUE TITLE
[BUG]: 8.8.0: fails to build with mbedtls

### ISSUE THREAD

**Issue by Kangie** [curl/curl#13748]
### I did this

```
./configure --prefix=/usr --build=x86_64-pc-linux-gnu --host=x86_64-pc-linux-gnu --mandir=/usr/share/man --infodir=/usr/share/info --datadir=/usr/share --sysconfdir=/etc --localstatedir=/var/lib --datarootdir=/usr/share --disable-dependency-tracking --disable-silent-rules --disable-static --docdir=/usr/share/doc/curl-8.8.0 --htmldir=/usr/share/doc/curl-8.8.0/html --with-sysroot=/ --libdir=/usr/lib64 --without-ca-fallback --with-ca-bundle=/etc/ssl/certs/ca-certificates.crt --without-gnutls --without-mbedtls --without-rustls --with-mbedtls --with-default-ssl-backend=mbedtls --enable-alt-svc --enable-basic-auth --enable-bearer-auth --enable-digest-auth --enable-kerberos-auth --enable-negotiate-auth --enable-aws --enable-dict --disable-ech --enable-file --enable-ftp --disable-gopher --enable-hsts --enable-http --enable-imap --disable-ldap --disable-ldaps --enable-ntlm --disable-ntlm-wb --enable-pop3 --enable-rt --enable-rtsp --disable-smb --without-libssh2 --enable-smtp --disable-telnet --enable-tftp --enable-tls-srp --enable-ares --enable-cookies --enable-dateparse --enable-dnsshuffle --enable-doh --enable-symbol-hiding --enable-http-auth --enable-ipv6 --enable-largefile --enable-manual --enable-mime --enable-netrc --enable-progress-meter --enable-proxy --enable-socketpair --disable-sspi --disable-static --enable-pthreads --enable-threaded-resolver --disable-versioned-symbols --without-amissl --without-bearssl --without-brotli --with-fish-functions-dir=/usr/share/fish/vendor_completions.d --with-nghttp2 --without-hyper --without-libidn2 --without-gssapi --without-libgsasl --with-libpsl --without-msh3 --without-nghttp3 --without-ngtcp2 --without-quiche --without-librtmp --without-schannel --without-secure-transport --without-test-caddy --without-test-httpd --without-test-nghttpx --disable-websockets --without-winidn --without-wolfssl --with-zlib --without-zstd --with-zsh-functions-dir=/usr/share/zsh/site-functions --with-test-nghttpx=/usr/bin/nghttpx

. . .
ibtool: compile:  x86_64-pc-linux-gnu-gcc -DHAVE_CONFIG_H -I/var/tmp/portage/net-misc/curl-8.8.0/work/curl-8.8.0/include -I../lib -I/var/tmp/portage/net-misc/curl-8.8.0/work/curl-8.8.0/lib -DBUILDING_LIBCURL -DCURL_HIDDEN_SYMBOLS -isystem /include -fvisibility=hidden -march=native -O2 -pipe -fdiagnostics-color=always -frecord-gcc-switches -ggdb3 -Werror-implicit-function-declaration -c /var/tmp/portage/net-misc/curl-8.8.0/work/curl-8.8.0/lib/vssh/wolfssh.c  -fPIC -DPIC -o vssh/.libs/libcurl_la-wolfssh.o
/var/tmp/portage/net-misc/curl-8.8.0/work/curl-8.8.0/lib/vtls/mbedtls.c: In function 'mbed_connect_step2':
/var/tmp/portage/net-misc/curl-8.8.0/work/curl-8.8.0/lib/vtls/mbedtls.c:936:15: error: implicit declaration of function 'mbedtls_ssl_get_ciphersuite_id_from_ssl'; did you mean 'mbedtls_ssl_get_ciphersuite_id'? [-Werror=implicit-function-declaration]
  936 |               mbedtls_ssl_get_ciphersuite_id_from_ssl(&backend->ssl);
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |               mbedtls_ssl_get_ciphersuite_id
cc1: some warnings being treated as errors
make[2]: *** [Makefile:3263: vtls/libcurl_la-mbedtls.lo] Error 1
```

[build.log.txt](https://github.com/curl/curl/files/15403163/build.log.txt)

### I expected the following

A successful build

### curl/libcurl version

curl 8.8.0
net-libs/mbedtls-2.28.7:0/7.14.1::gentoo

### operating system

Gentoo Linux

**Comment by bagder** [comment]
the new function use was brought by fba9afebba (@jan2000)

The mbedTLS has this funcion (mbedtls_ssl_get_ciphersuite_id_from_ssl) documented with a [single line](https://mbed-tls.readthedocs.io/projects/api/en/development/api/file/ssl_8h/#_CPPv439mbedtls_ssl_get_ciphersuite_id_from_sslPK19mbedtls_ssl_context) and there is no mention of when it was introduced...

**Comment by icing** [comment]
Could you try the following patch?

```diff
diff --git a/lib/vtls/mbedtls.c b/lib/vtls/mbedtls.c
index ec0b10dd9..98a4ea01b 100644
--- a/lib/vtls/mbedtls.c
+++ b/lib/vtls/mbedtls.c
@@ -902,8 +902,6 @@ mbed_connect_step2(struct Curl_cfilter *cf, struct Curl_easy *data)
     (struct mbed_ssl_backend_data *)connssl->backend;
   struct ssl_primary_config *conn_config = Curl_ssl_cf_get_primary_config(cf);
   const mbedtls_x509_crt *peercert;
-  char cipher_str[64];
-  uint16_t cipher_id;
 #ifndef CURL_DISABLE_PROXY
   const char * const pinnedpubkey = Curl_ssl_cf_is_proxy(cf)?
     data->set.str[STRING_SSL_PINNEDPUBLICKEY_PROXY]:
@@ -932,11 +930,18 @@ mbed_connect_step2(struct Curl_cfilter *cf, struct Curl_easy *data)
     return CURLE_SSL_CONNECT_ERROR;
   }

-  cipher_id = (uint16_t)
-              mbedtls_ssl_get_ciphersuite_id_from_ssl(&backend->ssl);
-  mbed_cipher_suite_get_str(cipher_id, cipher_str, sizeof(cipher_str), true);
-  infof(data, "mbedTLS: Handshake complete, cipher is %s", cipher_str);
-
+#if MBEDTLS_VERSION_NUMBER >= 0x03020000
+  {
+    char cipher_str[64];
+    uint16_t cipher_id;
+    cipher_id = (uint16_t)
+                mbedtls_ssl_get_ciphersuite_id_from_ssl(&backend->ssl);
+    mbed_cipher_suite_get_str(cipher_id, cipher_str, sizeof(cipher_str), true);
+    infof(data, "mbedTLS: Handshake complete, cipher is %s", cipher_str);
+  }
+#else
+  infof(data, "mbedTLS: Handshake complete");
+#endif
   ret = mbedtls_ssl_get_verify_result(&backend->ssl);

   if(!conn_config->verifyhost)
``

**Comment by icing** [comment]
Made this into #13749.

**Comment by jan2000** [comment]
Hmm, at time of writing I must have assumed availability of `mbedtls_ssl_get_ciphersuite_id_from_ssl()` was the same as `mbedtls_ssl_get_ciphersuite()` because of their similarity and spare documentation, and forgotten to check afterwards.

@icing 's fix seems to be the right one. Minor: the declaration and initialization of `cipher_id` can be combined now.

**Comment by bagder** [comment]
Fixed by  #13749, I missed to add the right keyword...
